main_pipeline.py

```diff
@@ -214,14 +226,26 @@ def process_keyframe(keyframe_path):
         keyframe_job = model.push(file_arg, detail_prompt)
         return keyframe_job
 
+    # only used if VILA is chosen
+    def process_scenes(scene_frames):
+        model = model_mapping[visual_detail]
+        file_arg = file_type[visual_detail](path=video_path)
+        keyframe_numbers = str(scene_frames)
+        keyframe_numbers = keyframe_numbers.replace("[", "").replace("]", "").replace(" ", "")
```

Comment:
this line seems like a hack. why are you doing string manipulation to get numbers?

Reply:
the input to VILA rn is a string that has the frame numbers in the form of comma-separated values. So, once we perform all the processing as lists, this converts it to a format that can be processed by VILA directly. I can make it cleaner though, will do that, but that step of list -> string is necessary to pass it into vila correctly.